subprojects/model-core/src/main/java/org/gradle/internal/instantiation/generator/InjectUtil.java

```diff
@@ -32,16 +32,16 @@
      * @param type the type to find the injectable constructor of.
      * @param reportAs errors are reported against this type, useful when the real type is generated, and we'd like to show the original type in messages instead.
      */
-    public static <T> ClassGenerator.GeneratedConstructor<T> selectConstructor(ClassGenerator.GeneratedClass<T> type, Class<?> reportAs) {
+    public static <T> int selectConstructor(ClassGenerator.GeneratedClass<T> type, Class<?> reportAs) {
```

Comment:
I think the Javadoc should be adjusted to clarify the return value of this method.